src/Generator/Common.php

```diff
@@ -95,11 +115,43 @@ public function __construct(\DBManager $db, StorageCommon $storageAdapter, $inse
     }
 
     /**
+     * @param string $moduleName
+     * @return int
+     */
+    public function getInsertCounterForModule($moduleName)
+    {
+        if (!array_key_exists($moduleName, $this->insertCounterModules)) {
+            return 0;
+        }
+
+        return $this->insertCounterModules[$moduleName];
+    }
+
+    /**
+     * @param string $moduleName
+     */
+    public function incrementInsertCounterForModule($moduleName)
+    {
+        $this->insertCounterModules[$moduleName] = $this->getInsertCounterForModule($moduleName) + 1;
+    }
+
+    /**
+     * @param Activity $activityStreamGenerator
+     */
+    public function setActivityStreamGenerator($activityStreamGenerator)
+    {
+        $this->activityStreamGenerator = $activityStreamGenerator;
+    }
+
+    /**
      * Data generator.
      *
      * @param int $number
      */
-    abstract public function generate($number);
+    public function generate($number)
```

Comment:
We need to better name for ```totalToGenerate``` setter, of return back abstract method and create setTotalToGenerate absract method